src/ram/StringConstant.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2013, 2014, Oracle and/or its affiliates. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file StringConstant.h
+ *
+ * Defines string constants
+ *
+ ************************************************************************/
+
+#pragma once
+
+#include "ram/Expression.h"
+#include "ram/Node.h"
+#include "souffle/RamTypes.h"
+#include "souffle/utility/StringUtil.h"
+#include <string>
+
+namespace souffle::ram {
+
+/**
+ * @class Constant
+ * @brief Represents a String Constant
+ *
+ */
+class StringConstant : public Expression {
+public:
+    StringConstant(std::string constant) : constant(constant) {}
+
+    /** @brief Get constant */
+    const std::string& getConstant() const {
+        return constant;
+    }
+
+    StringConstant* clone() const override {
+        return new StringConstant(constant);
+    }
+
+protected:
+    void print(std::ostream& os) const override {
+        os << "string(\"" << stringify(constant) << "\")";
```

Comment:
Is `stringify` needed?

Reply:
Yes - otherwise control codes are directly printed in the RAM output (e.g., `souffle <prog> --show=transformed-datalog`).